release/release.go

```diff
@@ -151,6 +151,32 @@ func CheckUpstreamRelease(ctx context.Context, client *github.Client, org, repo
 	return releases, nil
 }
 
+func KubernetesGoVersion(ctx context.Context, client *github.Client, version string) (string, error) {
+	var githubError *github.ErrorResponse
+
+	file, _, _, err := client.Repositories.GetContents(ctx, "kubernetes", "kubernetes", ".go-version", &github.RepositoryContentGetOptions{
+		Ref: version,
+	})
+
+	if err != nil {
+		if errors.As(err, &githubError) {
+			if githubError.Response.StatusCode == http.StatusNotFound {
+				return "", errors.New("failed to find .go-version file in given Kubernetes version")
+			}
+			return "", errors.New("unexpected GitHub API error")
```

Comment:
```suggestion
			return "", err
```

We don't want to lose the contents of the error here.